src/swarm/client/request/GetChannelsRequest.d

```diff
@@ -82,15 +82,16 @@ public scope class GetChannelsRequestTemplate ( Base : IRequest, Resources,
     {
         auto output = this.params.io_item.get_node_value();
 
-        do
-        {
-            this.reader.readArray(*this.resources.channel_buffer);
+        this.reader.readArray(*this.resources.channel_buffer);
 
+        while ( this.resources.channel_buffer.length );
```

Comment:
DMD has this nasty property to error when it shouldn't and not error when it should ;-)